tests/worker/test_workflow.py

```diff
@@ -789,6 +899,255 @@ async def test_workflow_cancel_multi(client: Client):
+        await asyncio.sleep(0.01)
+
+    async def wait_and_swallow(self, aw: Awaitable) -> None:
+        try:
+            await aw
+        except:
+            pass
+
+
+async def test_workflow_cancel_unsent(client: Client):
+    workflow_id = f"workflow-{uuid.uuid4()}"
+    async with new_worker(
+        client, CancelUnsentWorkflow, LongSleepWorkflow, activities=[wait_cancel]
+    ) as worker:
+        await client.execute_workflow(
+            CancelUnsentWorkflow.run,
+            id=workflow_id,
+            task_queue=worker.task_queue,
+        )
+    # Check history
+    resp = await client.service.get_workflow_execution_history(
+        GetWorkflowExecutionHistoryRequest(
+            namespace=client.namespace,
+            execution=WorkflowExecution(workflow_id=workflow_id),
+        )
+    )
+    found_timer = False
+    for event in resp.history.events:
+        # No activities or children scheduled
+        assert event.event_type is not EventType.EVENT_TYPE_ACTIVITY_TASK_SCHEDULED
+        assert (
+            event.event_type
+            is not EventType.EVENT_TYPE_START_CHILD_WORKFLOW_EXECUTION_INITIATED
+        )
+        # Make sure timer is just our 0.01 timer
+        if event.event_type is EventType.EVENT_TYPE_TIMER_STARTED:
+            assert (
+                event.timer_started_event_attributes.start_to_fire_timeout.ToMilliseconds()
+                == 10
+            )
+            found_timer = True
+    assert found_timer
+
+
+@workflow.defn
+class CancelMultipleAttemptWorkflow:
+    def __init__(self) -> None:
+        self._cancel_attempts = 0
+
+    @workflow.run
+    async def run(self, attempts_until_reraise: int) -> None:
+        timer_task: asyncio.Task = asyncio.create_task(asyncio.sleep(1000))
+        while True:
+            try:
+                await asyncio.shield(timer_task)
+            except asyncio.CancelledError:
+                self._cancel_attempts += 1
+                if self._cancel_attempts >= attempts_until_reraise:
+                    raise ApplicationError(
+                        f"cancelled after {self._cancel_attempts} attempts"
+                    )
+
+    @workflow.query
+    def cancel_attempts(self) -> int:
+        return self._cancel_attempts
+
+
+@workflow.defn
+class CancelMultipleAttemptParentWorkflow:
+    def __init__(self) -> None:
+        self._ready = False
+
+    @workflow.run
+    async def run(self, attempts_until_reraise: int) -> None:
+        self._child = await workflow.start_child_workflow(
+            CancelMultipleAttemptWorkflow.run,
+            attempts_until_reraise,
+            id=f"{workflow.info().workflow_id}_child",
+        )
+        self._ready = True
+        await self._child
+
+    @workflow.query
+    def ready(self) -> bool:
+        return self._ready
+
+    @workflow.signal
+    def cancel_child(self) -> None:
+        self._child.cancel()
+
+
+@pytest.mark.skip(reason="repeated cancellation not sent")
```

Comment:
I'd just change the test and assert that cancellation is only sent once as intended.

Reply:
Considered it, but that means I have to have a timeout-based test which is unfortunate (basically I have to wait long enough to confirm that cancellation is not resent somehow). I considered just deleting this test altogether.

Reply:
Could just delete it, server behavior isn't going to change

Reply:
Deleted